### PR TITLE
Documents STM32F0 family RCC

### PR DESCRIPTION
**Pull request by octronics**
Not all registers nor fields are covered. Will continue working on.

### CODE REVIEW

**Comment by octronics** [comment]
Have you a preference for naming values?
Example:
- `DivideBy2`, what i have used. Give a `divide_by2()` function.
- `DividedBy2`, better for read only value. not so great for writable value.
- Use both `DivideBy2` for writing and `DividedBy2` for reading. But gives more works and duplication.
- `Div2`, you use it on other values. Short but make code less clear.

**Comment by octronics** [comment]
CI catch an error that doesn't appear on my computer when I run `make patch` to test the change. I think it can be caused by  invalid  .deps file. Is it possible to run makedeps before svdpatch each time `make patch`? Otherwise when I change branch I sometime have to `rm .deps/ -R`.

**Comment by octronics** [comment]
All devices on the F0 family have Touch sensing except F0x0 devices. If I describe the field in `rcc_f0.yaml` `make patch` fail with `Could not find RCC:AHBENR.TSCEN`. It is normal and really helpful when writing descriptions. Do I need to not describe this field on peripheral and instead describe it on devices files? But if I do that, there will be 3 places to maintain instead on just one. Or does a way to tag a field as a not always present field exist?

**Comment by adamgreig** [comment]
Thanks for the hard work!

> Have you a preference for naming values?

I think `DivideBy2` (for both r+w) is the nicest of your options, but for consistency with everything that already exists please stick with `Div2`.

> Is it possible to run makedeps before svdpatch each time make patch? 

The Makefile should automatically rerun makedeps for each device whenever the respective device YAML file changes; do you know what changed in your case that didn't work?

> All devices on the F0 family have Touch sensing except F0x0 devices.

I appreciate that it's annoying but I think the best option for this scenario is a small `rcc_f0_tsc.yaml` which you can include in all the non-F0x0 devices. Checking that all described peripherals at least exist in the SVD is one of the few things we can automatically check so I'd rather keep it in.

File: peripherals/rcc/rcc_f0.yaml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      HSIDividedBy2: [0, "HSI divided by 2 selected as PLL input clock"]
+      HSIDividedByPREDIV: [1, "HSI divided by PREDIV selected as PLL input clock"]
+      HSEDividedByPREDIV: [2, "HSE divided by PREDIV selected as PLL input clock"]
+      HSI48DividedByPREDIV: [3, "HSI48 divided by PREDIV selected as PLL input clock"]
+    PPRE:
+      NotDivided: [0, "HCLK not divided"] # Same for [0, 8]
+      DividedBy2: [4, "HCLK divided by 2"]
+      DividedBy4: [5, "HCLK divided by 4"]
+      DividedBy8: [6, "HCLK divided by 8"]
+      DividedBy16: [7, "HCLK divided by 16"]
```
[the remark this answers is not in the document]
Comment (adamgreig) [reply peripherals/rcc/rcc_f0.yaml]: :+1:

File: peripherals/rcc/rcc_f0.yaml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        Clear: [1, "Clear HSIRDYF flag"]
+    LSERDYC:
+      _write:
+        Clear: [1, "Clear LSERDYF flag"]
+    LSIRDYC:
+      _write:
+        Clear: [1, "Clear LSIRDYF flag"]
+    HSI48RDYIE:
+      Disable: [0, "HSI48 ready interrupt disabled"]
+      Enable: [1, "HSI48 ready interrupt enabled"]
```
[the remark this answers is not in the document]
Comment (adamgreig) [reply peripherals/rcc/rcc_f0.yaml]: :+1:

File: peripherals/rcc/rcc_f0.yaml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    LSERDYIE:
+      Disable: [0, "LSE ready interrupt disabled"]
+      Enable: [1, "LSE ready interrupt enabled"]
+    LSIRDYIE:
+      Disable: [0, "LSI ready interrupt disabled"]
+      Enable: [1, "LSI ready interrupt enabled"]
+    CSSF:
+      _read:
+        NoInterrupt: [0, "No clock security interrupt caused by HSE clock failure"]
+        Interrupt: [1, "Clock security interrupt caused by HSE clock failure"]
```
[the remark this answers is not in the document]
Comment (adamgreig) [reply peripherals/rcc/rcc_f0.yaml]: Yes, that would be more consistent.

File: peripherals/rcc/rcc_f0.yaml [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    PLLRDY:
+      _read:
+        Unlocked: [0, "PLL unlocked"]
+        Locked: [1, "PLL locked"]
+    PLLON:
+      "Off": [0, "PLL off"]
+      "On":  [1, "PLL on" ]
+    CSSON:
+      Disable: [0, "Clock security system disabled (clock detector OFF)"]
+      Enable: [1, "Clock security system enable (clock detector ON if the HSE is ready, OFF if not)"]
```
[the remark this answers is not in the document]
Comment (adamgreig) [reply peripherals/rcc/rcc_f0.yaml]: I think `On` and `Off` are fine for `*ON` fields, so best use that here too.

File: peripherals/rcc/rcc_f0.yaml [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+      MultiplyBy13: [11, "PLL input clock x13"]
+      MultiplyBy14: [12, "PLL input clock x14"]
+      MultiplyBy15: [13, "PLL input clock x15"]
+      MultiplyBy16: [14, "PLL input clock x16"]
+      MultiplyBy16: [15, "PLL input clock x16"]
+    PLLSRC:
+      HSIDividedBy2: [0, "HSI divided by 2 selected as PLL input clock"]
+      HSIDividedByPREDIV: [1, "HSI divided by PREDIV selected as PLL input clock"]
+      HSEDividedByPREDIV: [2, "HSE divided by PREDIV selected as PLL input clock"]
+      HSI48DividedByPREDIV: [3, "HSI48 divided by PREDIV selected as PLL input clock"]
```
[the remark this answers is not in the document]
Comment (adamgreig) [reply peripherals/rcc/rcc_f0.yaml]: In principle I like this, but I worry about how many places we'd have to do that to be consistent and how much extra work that would be. In this case I wonder if `HSIDiv2`, `HSEDivPREDIV`, etc, might actually be better. They'd get `hsidiv2()` and `is_hsidiv2()` I guess, which seems OK.

It's annoying though as I was about to say I quite like `DividedBy2` etc!

**Comment by adamgreig** [comment]
> What do you think about renaming those Div2 to something more expressive? We write code once but need to read it multiple times. Abbreviations make this more difficult. I know this will be breaking changes. Perhaps on a future major release.

You have a good point and it might even be worth doing this in the future. I don't think svd2rust will allow multiple names per value or we could just have both `Div2` and `DividedBy2`; shame.

That said I think `Div2` is still clear and is quicker to read/scan than `DividedBy2` in code, in addition to getting around the formatting issues you mentioned on `PLLSRC`.

For now I'd be happy for you to keep `DividedBy2` here _or_ swap to `Div2`; we can consider moving everything one way or the other for a future major release.

**Comment by octronics** [comment]
While implementing the clocks module of a HAL crate, I found that `DivideBy2` given an `DIVIDEBY2` constant. `DIVIDE_BY_2` is much more readable. So I use explicit `_` separator to increase readability. 
Using short `Div` prevent the `Divided` and `Divide` dilemma.

**Comment by adamgreig** [comment]
Fixes look good, thanks. My only hesitation now is `Div_2` vs `Div2` - I can see it being better for `DIVIDE_BY_2` but not sure about it with the shorter `DIV`: `DIV_2` doesn't seem any more readable than `DIV2`. If you want to stick with it I'll go ahead and merge this, though.

**Comment by octronics** [comment]
I prefer `Div_2`. When you a have an enum full of `DIV2`, `DIV128` and the like it became hard to read them.

**Comment by adamgreig** [comment]
Okay! Could you fix the merge conflicts and then I'll merge this?

**Comment by octronics** [comment]
Rebase on top of your master

**Comment by adamgreig** [comment]
Thank you!

**Comment by octronics** [comment]
Thank you too!

**Pull request merged**